scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,217 @@
+        this.index = new int[] {index};
+        this.start = new Label[] {new Label()};
+        this.end = new Label[] {new Label()};
+        this.variableName = null;
+
+        try {
+            Field flagsField = Label.class.getDeclaredField("flags");
+            Field bytecodeOffsetField = Label.class.getDeclaredField("bytecodeOffset");
+            Field FLAG_RESOLVED_FIELD = Label.class.getDeclaredField("FLAG_RESOLVED");
+
+            flagsField.setAccessible(true);
+            bytecodeOffsetField.setAccessible(true);
+            FLAG_RESOLVED_FIELD.setAccessible(true);
+
+            int FLAG_RESOLVED = (Integer) FLAG_RESOLVED_FIELD.get(null);
+
+            short flagsStart = (Short) flagsField.get(start[0]);
+            short flagsEnd = (Short) flagsField.get(end[0]);
+            flagsStart |= FLAG_RESOLVED;
+            flagsEnd |= FLAG_RESOLVED;
+
+            flagsField.set(start[0], flagsStart);
+            bytecodeOffsetField.set(start[0], scopeStart);
+
+            flagsField.set(end[0], flagsEnd);
+            bytecodeOffsetField.set(end[0], scopeStart + scopeLength);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
     }
 
+    /**
+     * Returns the bytecode offset to the start of the scope.
+     * @return The bytecode offset to the start of the scope
+     */
+    public int getScopeStart() {
+        try {
+            return start[0].getOffset();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(start));
+            return -1;
+        }
+    }
 
     /**
-     * Returns whether this {@link LocalLocation} equals <code>o</code>; a
-     * slightly faster variant of {@link #equals(Object)} for when the argument
-     * is statically known to be another nonnull {@link LocalLocation}.
+     * Returns the length of the scope (in bytes).
+     * @return The length of the scope (in bytes)
      */
-    public boolean equals(LocalLocation l) {
-        return index == l.index && scopeStart == l.scopeStart
-                && scopeLength == l.scopeLength &&
-                (varName==null || varName.equals(l.varName)) &&
-                varIndex==l.varIndex;
+    public int getScopeLength() {
+        // Should this be end[0] instead?
+        try {
+            return end[end.length - 1].getOffset() - getScopeStart();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(end));
+            return -1;
+        }
     }
 
     /**
-     * This {@link LocalLocation} equals <code>o</code> if and only if
-     * <code>o</code> is another nonnull {@link LocalLocation} and
-     * <code>this</code> and <code>o</code> have equal {@link #index},
-     * {@link #scopeStart}, and {@link #scopeLength}.
+     * Returns the local variable index.
+     * @return The local variable index
      */
+    public int getVarIndex() {
+        return index[0];
```

Comment:
Is it an invariant that all elements of `index` are equal?
If it's not an invariant, should it be checked here?  What goes wrong if the array values aren't all the same?